site/app/controllers/student/SubmissionController.php

```diff
@@ -114,7 +114,7 @@ public function showHomeworkPage($gradeable_id, $gradeable_version = null) {
             return $verify_permissions;
         }
 
-        if ($gradeable->isLocked($this->core->getUser()->getId())) {
+        if ($gradeable->isLocked($this->core->getUser()->getId()) && ($this->core->getUser()->accessGrading() === false)) {
```

Comment:
These parentheses are unnecessary
```suggestion
        if ($gradeable->isLocked($this->core->getUser()->getId()) && $this->core->getUser()->accessGrading() === false) {
```